src/components/designer/taro/actions/NewAddressModal.tsx

```diff
@@ -0,0 +1,185 @@
+  const { hideNewAddress } = useStoreActions(s => s.modals);
+
+  //taro model
+  const { getNewAddress } = useStoreActions(s => s.taro);
+
+  //component state
+  const [selectedAmount, setSelectedAmount] = useState('10');
+  const [selectedGenesisBootstrapInfo, setSelectedGenesisBootstrapInfo] = useState('');
+  const [selectedBalance, setSelectedBalance] = useState<PTARO.TaroBalance>();
+  const [taroAddress, setTaroAddress] = useState('');
+
+  //component local variables
+  const [form] = Form.useForm();
+  const thisTaroNode = network.nodes.taro.find(
+    node => node.name === nodeName,
+  ) as TarodNode;
+  const otherTaroNodes: TaroNode[] = network.nodes.taro.filter(
+    node => node.name !== nodeName,
+  );
+
+  //When polar is first opened, we need to populate the state with the lightning node data
+  useEffect(() => syncChart(network), []);
+
+  const handleSelectedBalance = (value: PTARO.TaroBalance) => {
+    setSelectedBalance(value);
+    setSelectedGenesisBootstrapInfo(value.genesisBootstrapInfo);
+    form.setFieldsValue({
+      genesisBootstrapInfo: value.genesisBootstrapInfo,
+      amount: value.type == PTARO.TARO_ASSET_TYPE.COLLECTIBLE ? 1 : 10,
+    });
+  };
+
+  //submit
+  const newAddressAsync = useAsyncCallback(async (payload: NewAddressPayload) => {
+    try {
+      const res = await getNewAddress(payload);
+      setTaroAddress(res.encoded);
+    } catch (error: any) {
+      notify({ message: l('submitError'), error });
+    }
+  });
+
+  const handleSubmit = (values: { genesisBootstrapInfo: string; amount: string }) => {
+    if (selectedGenesisBootstrapInfo) {
+      const payload: NewAddressPayload = {
+        node: thisTaroNode,
+        genesisBootstrapInfo: values.genesisBootstrapInfo,
+        amount: values.amount,
+      };
+      newAddressAsync.execute(payload);
+    }
+  };
+
+  const handleCopy = () => {
+    message.success(l('copied', { address: ellipseInner(taroAddress, 10, 10) }), 2);
+    hideNewAddress();
+  };
+
+  let cmp: ReactNode;
+  if (!taroAddress) {
+    cmp = (
+      <>
+        <Form
+          form={form}
+          layout="vertical"
+          colon={false}
+          initialValues={{
+            amount: '10',
+            node: '',
+            genesisBootstrapInfo: '',
+          }}
+          onFinish={handleSubmit}
+        >
+          <Form.Item name="genesisBootstrapInfo" label={l('genesisBootstrapInfo')}>
+            <Input.TextArea
+              rows={5}
+              placeholder={l('genesisBootstrapInfo')}
+              onChange={e => setSelectedGenesisBootstrapInfo(e.target.value)}
+            />
+          </Form.Item>
+          <Styled.Spacer />
+
+          <Styled.TaroDataSelect
+            name="node"
+            label={l('selectBalance')}
+            taroNetworkNodes={otherTaroNodes}
+            selectBalances
+            onChange={v => handleSelectedBalance(v?.valueOf() as PTARO.TaroBalance)}
```

Comment:
Using `valueOf()` is a bit of a code smell in Typescript, similar to using `any`. We should try to avoid it unless absolutely necessary. I would suggest updating the `TaroDataSelect.onChange` prop to use the proper type, then you wouldn't need to do this type casting.

Reply:
Done. Yea my typescript game is weak. Lots to learn here.